cpp/ql/test/library-tests/dataflow/dataflow-tests/test_diff.expected

```diff
@@ -1,3 +1,4 @@
+| clang.cpp:12:9:12:20 | clang.cpp:18:8:18:19 | AST only |
```

Comment:
That doesn't look good to me.

Reply:
It looks like we currently don't allow data flow through chi nodes at all. Would it be more consistent with AST data flow to allow flow through the "total" operand of chi nodes? Would that fix this problem?

Reply:
It would fix this case, but it would also introduce false positives. I don't think it would be an improvement overall.

Reply:
I've been thinking about this a bit more, and I wonder if we should differentiate Chi nodes writing to all aliased memory from Chi nodes writing to specific virtual variables - flow through the former is much more likely to introduce false positives

Reply:
That sounds right. @dbartol, I think you had something similar in mind?

Reply:
After our meeting yesterday I understand that we have to take this step backwards in order to take a step forward in another PR.